src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/adapters/InMemoryDataAdapter.java

```diff
@@ -0,0 +1,35 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.adapters;
+
+import org.openkilda.DataAdapter;
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchId;
+import org.openkilda.model.SwitchProperties;
+
+import lombok.Value;
+
+import java.util.Collection;
+import java.util.Map;
+
+@Value
+public class InMemoryDataAdapter implements DataAdapter {
+
+    Collection<FlowPath> flowPaths;
```

Comment:
For example we need to create flow with protected path. In this case `flowPath` will contain 4 paths. 2 primary and 2 protected. For protected paths we must NOT install ingress rules. But how to understand that path in collection is protected without Flow object? FlowPath model doesn't contains such infomation

Reply:
Reworked adapter.